File: backend/src/main/java/com/apocalypse/thefall/service/character/rules/perk/PerkRule.java
package com.apocalypse.thefall.service.character.rules.perk;

import com.apocalypse.thefall.entity.character.Character;
import com.apocalypse.thefall.entity.character.stats.Perk;
import com.apocalypse.thefall.entity.character.stats.SkillEnum;
import com.apocalypse.thefall.entity.character.stats.SpecialEnum;

import java.util.Map;

public interface PerkRule {
    boolean apply(Perk perk, Character character, Map<SpecialEnum, Integer> specialMap, Map<SkillEnum, Integer> skillValues, Integer rank);
}

File: backend/src/main/java/com/apocalypse/thefall/service/character/rules/perk/PerkRule.kt
package com.apocalypse.thefall.service.character.rules.perk

import com.apocalypse.thefall.entity.character.Character
import com.apocalypse.thefall.entity.character.stats.Perk
import com.apocalypse.thefall.entity.character.stats.enums.SkillEnum
import com.apocalypse.thefall.entity.character.stats.enums.SpecialEnum

interface PerkRule {
    fun apply(
        perk: Perk,
        character: Character,
        specialMap: Map<SpecialEnum, Int>,
        skillValues: Map<SkillEnum, Int>,
        rank: Int
    ): Boolean
}
